.github/workflows/ci.yml

```diff
@@ -0,0 +1,40 @@
+name: CI
+
+on:
+    push:
+        branches: main
+        paths: ['**.cpp', '**.h', '**CMakeLists.txt'] 
+    pull_request:
+        branches: main
+        paths: ['**.cpp', '**.h', '**CMakeLists.txt']
+
+jobs:
+    build-and-test:
+        runs-on: ubuntu-latest
+        steps:
+            - name: Install Java
+              uses: actions/setup-java@v4
+              with:
+                distribution: 'oracle' # See 'Supported distributions' for available options
+                java-version: '17'
+            
+            - name: Clone Minecraft_Tools
+              run: git clone https://github.com/nhatdongdang/Minecraft_Tools
+
+            - name: Start Spigot Server
+              working-directory: ./Minecraft_Tools/server
+              run: nohup java -Xms512M -Xmx1024M -jar -DIReallyKnowWhatIAmDoingISwear spigot-1.19.4.jar nogui -o true & sleep 45
+
+            - name: Clone mcpp
+              uses: actions/checkout@v4
+
+            - name: Build file
+              run: cmake -Bbuild
+
+            - name: Run  local_tests
+              working-directory: ./build
+              run: make local_tests && ./test/local_tests
+
+            - name: Run test_suite
+              working-directory: ./build
+              run: make test_suite && ./test/test_suite
```

Comment:
`test_suite` already includes local tests, and this should probably be run using `ctest` unless there is any issue with that.

Reply:
Thank you for your feedback. I have already fixed the issue, and the test suite is now set up to run using ctest. Please let me know if you encounter any further issues.

Reply:
Thank you for your feedback. I have already fixed the issue, and the test suite is now set up to run using ctest. Please let me know if you encounter any further issues.